core/templates/components/code-mirror/codemirror-mergeview.component.ts

```diff
@@ -27,14 +28,27 @@ import { downgradeComponent } from '@angular/upgrade/static';
 export class CodemirrorMergeviewComponent implements
   AfterViewInit, OnInit, OnChanges {
   @Input() options = {};
-  @Input() leftValue;
-  @Input() rightValue;
-  codeMirrorInstance: CodeMirror.MergeView.MergeViewEditor;
-  constructor(private elementRef: ElementRef, private ngZone: NgZone) { }
+  // Below properties have been assigned as empty string if undefined.
+  // See function 'ngAfterViewInit' for reference.
+  @Input() leftValue: string | undefined;
+  @Input() rightValue: string | undefined;
+  // 'ngOnChanges' sometimes runs before the view has been initialized,
+  // to cater this we are checking it be to not undefined.
+  codeMirrorInstance!: CodeMirror.MergeView.MergeViewEditor;
+
+  get getCodeMirrorInstance(): CodeMirror.MergeView.MergeViewEditor {
+    return this.codeMirrorInstance;
+  }
```

Comment:
Why is this needed? Can't you just `spyOn(window.CodeMirror, 'MergeView').and.callFake((...args) => mockObject) `?